test/core/templaters/jinja_test.py

```diff
@@ -866,7 +866,8 @@ def test__templater_jinja_slice_template(test, result):
 
 
 def _statement(*args, **kwargs):
-    return "_statement"
+    # NOTE: The standard dbt statement() call returns nothing.
```

Comment:
Should this be a _new_ test case rather than changing the existing one? In other words, just because dbt's `statement()` call returns nothing, it seems like it should still be valid for some other Jinja or dbt macro to return something. As written, I worry we may be breaking something or creating an opening for future issues.

Reply:
Good question 🤔 . I think I'll keep this test the same, but I will add a more general test for jinja `call` blocks 👍

Reply:
That was a *great* call. It was waaaay more tricky than expected to do that, but I'm confident that the result is much better. I now understand a bit more why it was originally done the way you did.

Turns out that the conflict was that the lexer assumes that blocks must have zero length. The tracer puts any `call` results in the `block_start`. When we renamed the `block_start` to a `templated` section - that made the code _work_ but block counting in the lexer would fail. I've update the code now to allow `block_start` sections to have length, and then block count effectively when they do.

Reply:
In light of the discussion above, are you planning to undo the change of `"_statement"` -> `""`?

Reply:
(Having a bit of trouble comprehending the test additions vs changes.)

Reply:
In this case I don't think so. The example query further down won't parse properly if we do that. What I wanted to do instead was have another test case where we do a `{% call something_else() %}`, rather than just `statement()` (which isn't something supported out of the box in jinja - or supported by the context of tools like airflow).

In short:
- `{% call statement(...) %}` I think should always return nothing, but just so the test cases are representative of likely use cases.
- We do _also_ need test cases which also cover situations where `{% call xyz() %}` does return a string with length, which is why I assed the jinja tests with `JINJA_MACRO_CALL_SQL` and the additional test cases in `lexer_test.py`.